utils/safeList.js

```diff
@@ -0,0 +1,286 @@
+const xml2js = require('xml2js');
+
+const recoverableErrors = new Set([
+    'TimestampParserError',
+]);
+
+const NO_VALUE = Symbol('noValue');
+
+/**
+ * @typedef {Object} ExtractResult
+ * @property {any} value - extracted value
+ * @property {Error|undefined} error - error if any
+ */
+
+/**
+ * @typedef {function} Extractor
+ * @param {string} key - key to extract
+ * @param {object} data - data to extract from
+ * @returns {ExtractResult} - extracted value
+ */
+
+/**
+ * extractKeys - extract keys from data using the provided extractors
+ *
+ * @param {object} data - data to extract keys from
+ * @param {object<string, Extractor>} keys - Mapping of keys to extractors
```

Comment:
Since `Extractor` is a function type, we could want to document here what the function should take as arguments, and what it should return.